### PR TITLE
040 login system

### PR DESCRIPTION
**Pull request by yanjundev**

### CODE REVIEW

File: src/main/java/org/kainos/ea/models/UserResponse.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+public class UserResponse {
+
+    String email;
+
+    String password;
+
+    @JsonCreator
+    public UserResponse( @JsonProperty("email") String email,
+                         @JsonProperty("password") String password ) {
```
Comment (shaunganley) [inline src/main/java/org/kainos/ea/models/UserResponse.java]: not sure you should have a password in a response object as you'll never return it from the API?

File: src/main/java/org/kainos/ea/resources/AuthenticationService.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+            return Response.status( HttpStatus.UNAUTHORIZED_401 ).entity( e.getMessage() ).build();
+        }
+    }
+
+    @POST
+    @Path("/signout")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response removeUserToken( String token ) {
```
Comment (shaunganley) [inline src/main/java/org/kainos/ea/resources/AuthenticationService.java]: usually you wouldn't need a token to be able to sign out

File: src/main/java/org/kainos/ea/service/UserService.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+    public String authenticateUser( String email, String password ) throws DatabaseConnectionException, SQLException, InvalidUserCredentialsException {
+
+        int id = checkCredentials( email, password );
+
+        if ( id > 0 ) {
+
+            String token = JwtToken.generateToken( email );
+
+            if ( token != null ) {
```
Comment (shaunganley) [inline src/main/java/org/kainos/ea/service/UserService.java]: should there be an else statement if this is null?

**Review by shaunganley — CHANGES_REQUESTED** [review]
great work on this, few small comments

File: src/main/java/org/kainos/ea/models/UserResponse.java (hunk 1 above)
Comment (yanjundev) [reply src/main/java/org/kainos/ea/models/UserResponse.java]: that's my bad naming sense

File: src/main/java/org/kainos/ea/resources/AuthenticationService.java (hunk 2 above)
Comment (yanjundev) [reply src/main/java/org/kainos/ea/resources/AuthenticationService.java]: changed sending token in body to "Authorization" header

**Review by shaunganley — APPROVED** [review]

**Review by matthewknox99 — APPROVED** [review]

**Pull request merged**